### PR TITLE
Console list all wells command

### PR DESCRIPTION
**Pull request by steinsiv**
## Description
Adding option to list all wells

## Type of change
* Enhancement of existing functionality

## Impacted Areas in Application
* /Console

## Checklist:

*Communication*
* [ ] PR is related to an issue
* [ ] I have made corresponding changes to the documentation

*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass
* [ ] New code is covered by passing tests

### CODE REVIEW

File: Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                {
+                    Uid = "",
+                    Name = "",
+                    Field = "",
+                    Country = "",
+                    Operator = ""
+                }.AsSingletonList()
+            };
+            var result = await witsmlClient.GetFromStoreAsync(wellsQuery, OptionsIn.Requested);
+            foreach (var res in result.Wells)
```
Comment (HakonHesselberg) [inline Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs]: res -> well?

**Review by HakonHesselberg — APPROVED** [review]
Code looks good to me.

Code not tested, recommend another reviewer.

**Comment by lindsve** [comment]
Code looks good to me as well.

But I'm wondering about the utility of it when it lists _all_ wells from a server as that could be thousands

**Review by lindsve — CHANGES_REQUESTED** [review]
If a command for listing all wells is useful, then it would at least require adding paging capabilities. I can't seem to see the usefulness of listing 100/1000s of wells in the console.

One solution could be to list ~20 wells, then wait for a keypress and then list the next 20.

But again, I struggle to find the utility of it

File: Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs (hunk 1 above)
Comment (steinsiv) [reply Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs]: fixed!

**Comment by steinsiv** [comment]
It was implemented more as an exercise to expand on the Console app.
We could perhaps modify this command to filter the results to be something like:
```
USAGE:
    dotnet run -- list [OPTIONS] <COMMAND>

OPTIONS:
    -h, --help    Prints help information
                                         
COMMANDS:
    wells <KEYWORD> <LIMIT>               List wells, based on <keyword>. Return max <limit> results
    wellbores                             List active wellbores               
    logs <WELL_UID> <WELLBORE_UID>        List logs within a well/wellbore    
    tubulars <WELL_UID> <WELLBORE_UID>    List tubulars within a well/wellbore
```
If there is no use for it we should not merge it, I agree.

File: Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            var result = await witsmlClient.GetFromStoreAsync(wellsQuery, OptionsIn.Requested);
+            foreach (var well in result.Wells)
+            {
+                yield return new WitsmlWell
+                {
+                    Uid = well.Uid,
+                    Name = well.Name,
+                    Field = well.Field != null ? well.Field : "",
+                    Country = well.Country != null ? well.Country : "",
+                    Operator = well.Operator != null ? well.Operator : ""
```
Comment (stigrus) [inline Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs]: Believe these three can be written using [null-coalescing operator](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator) as:
```
Field = well.Field ?? "",
Country = well.Country ?? "",
Operator = well.Operator ?? ""
```

**Review by stigrus — CHANGES_REQUESTED** [review]
One small comment

File: Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs (hunk 2 above)
Comment (steinsiv) [reply Src/WitsmlExplorer.Console/ListCommands/ListWellsCommand.cs]: Thanks! 👍

**Review by stigrus — APPROVED** [review]

**Comment by steinsiv** [comment]
Reverting to implement changes for later

**Pull request closed without merging**